contrib/nydusd-http-server/src/main.rs

```diff
@@ -0,0 +1,236 @@
+#[macro_use]
+extern crate rocket;
+extern crate http_range;
+
+use rocket::fs::FileServer;
+use std::env;
+
+// If we wanted or needed to serve files manually, we'd use `NamedFile`. Always
+// prefer to use `FileServer`!
+mod getblob {
+    use http_range::HttpRange;
+    use rocket::fs::NamedFile;
+    use rocket::http::Status;
+    use rocket::request::{self, FromRequest, Outcome};
+    use rocket::response::{self, stream::ReaderStream, Responder};
+    use rocket::Request;
+    use rocket::Response;
+    use std::io::{Read, Seek};
+    use std::path::{Path, PathBuf};
+    use std::{env, fs, io};
+
+    #[derive(Debug)]
+    pub struct HeaderData {
+        _host: String,
+        range: String,
+        _authstr: String,
+    }
+
+    #[rocket::async_trait]
+    impl<'r> FromRequest<'r> for HeaderData {
+        type Error = Status;
+
+        async fn from_request(req: &'r Request<'_>) -> request::Outcome<HeaderData, Self::Error> {
+            let host = match req.headers().get_one("Host") {
+                Some(h) => h.to_string(),
+                None => "".to_string(),
+            };
+            let rangestr = match req.headers().get_one("Range") {
+                Some(h) => h.to_string(),
+                None => "".to_string(),
+            };
+            let authstr = match req.headers().get_one("Authorization") {
+                Some(h) => h.to_string(),
+                None => "".to_string(),
+            };
+            Outcome::Success(HeaderData {
+                _host: host,
+                range: rangestr,
+                _authstr: authstr,
+            })
+        }
+    }
+
+    pub struct Layerinfo(NamedFile, String);
+    impl<'r> Responder<'r, 'static> for Layerinfo {
+        fn respond_to(self, _req: &'r Request<'_>) -> response::Result<'static> {
```

Comment:
Seems this method sends the whole file data to the client, use [this](https://api.rocket.rs/master/rocket/fs/struct.NamedFile.html#examples-4) should be better.